app/src/main/java/cz/muni/fi/pv256/movio2/uco_410434/adapter/FilmsAdapter.java

```diff
@@ -0,0 +1,138 @@
+                final View titleContainer = holder.mView.findViewById(R.id.title_container);
+
+                Log.i(TAG, "Binding film " + currentFilm.toString());
+                filmTitle.setText(currentFilm.getTitle());
+                filmRating.setRating((float) currentFilm.getVoteAverage());
+                filmImage.setImageDrawable(null);
+                fallbackBGColor(titleContainer, filmImage);
+                if (currentFilm.getBackdropPath() != null) {
+                    holder.mView.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED);
+                    Log.i(TAG, "Loading film " + currentFilm.getTitle() + ", image " + currentFilm.getBackdropPath());
+                    Picasso
+                            .with(filmListFragment.getContext())
+                            .load(currentFilm.getBackdropURL())
+                            .resize(holder.mView.getMeasuredWidth(), holder.mView.getMeasuredHeight())
+                            .placeholder(R.color.indigo_teal_primary)
+                            .into(filmImage, new Callback() {
+                                @Override
+                                public void onSuccess() {
+                                    Log.i(TAG, "Done loading film " + currentFilm.getTitle() + ", image " + currentFilm.getBackdropPath());
+                                    FilmPosterColorUtil.setBackgroundColor(filmListFragment.getContext(), filmImage.getDrawable(), titleContainer);
+                                }
+
+                                @Override
+                                public void onError() {
+                                    Log.e(TAG, "Could not load image " + currentFilm.getBackdropPath());
+                                    fallbackBGColor(titleContainer, filmImage);
+                                }
+                            });
+                }
+
+                holder.mView.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        filmListFragment.getFilmSelectListener().onFilmSelect(currentFilm);
+                    }
+                });
+                break;
+            case CATEGORY:
+                TextView textView = holder.mView.findViewById(R.id.category_text);
+                Category currentCategory = (Category) items[position];
+                textView.setText(currentCategory.getName());
+                break;
+        }
+    }
+
+    private void fallbackBGColor(View titleContainer, ImageView filmImage) {
+        titleContainer.setBackgroundColor(filmListFragment.getResources().getColor(R.color.indigo_teal_primary_dark));
+        filmImage.setBackgroundColor(filmListFragment.getResources().getColor(R.color.indigo_teal_primary));
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (items[position] instanceof Film) return FILM;
```

Comment:
je lepsie vzdy pouzivat zatvorky - prevencia pred chybami v buducnosti. 
alebo si mozes ulozit type do objektu a skratit tuto metodu na 
"return items[position].getType()" a nejaky nullcheck

Reply:
Opravím do task 8, neviem jak mi to vypadlo, raz si mi to už písala